server/build_event_protocol/build_event_handler/build_event_handler.go

```diff
@@ -1080,16 +1085,34 @@ func LookupInvocation(env environment.Env, ctx context.Context, iid string) (*in
 				log.Warningf("Error reading proto from log: %s", err)
 				return err
 			}
-			if ti.RedactionFlags&redact.RedactionFlagStandardRedactions == 0 {
+			if redactor != nil {
 				if err := redactor.RedactAPIKeysWithSlowRegexp(ctx, event.BuildEvent); err != nil {
 					return err
 				}
 				redactor.RedactMetadata(event.BuildEvent)
 			}
-			parser.ParseEvent(event)
+			events = append(events, event)
+			switch p := event.BuildEvent.Payload.(type) {
+			case *build_event_stream.BuildEvent_Progress:
+				if screenWriter != nil {
+					screenWriter.Write([]byte(p.Progress.Stderr))
+					screenWriter.Write([]byte(p.Progress.Stdout))
+				}
+				// Now that we've updated our screenwriter, zero out
+				// progress output in the event so they don't eat up
+				// memory.
+				p.Progress.Stderr = ""
+				p.Progress.Stdout = ""
+			case *build_event_stream.BuildEvent_StructuredCommandLine:
+				structuredCommandLines = append(structuredCommandLines, p.StructuredCommandLine)
+			}
 		}
 		invocationMu.Lock()
-		parser.FillInvocation(invocation)
```

Comment:
If I'm following correctly, the idea here is that since we already do this when before writing the invocation to the DB, all the invocation fields should already be filled by TableInvocationToProto.

For in-progress invocations though, I think we flush to blobstore more frequently than we flush to mysql. I think we only update the mysql row like 3 times in total throughout the invocation, but blobstore is updated every time we hit the chunk size limit in protofile.go

So I think if we remove this logic to go through the event parser, invocations might feel a bit more "stale"

Reply:
I have now addressed this by ensuring all the "starting events" (direct children of the `BuildStarted` event that contain metadata) are processed prior to the second db write, which happens very early during processing. The goal here is to eventually not read the events protofile at all during invocation lookup.

Reply:
This looks like a reasonably safe change but would you mind putting it behind a flag in case we find that it is causing the metadata to be written later than we expect and we need a way to go back to the old behavior?

Reply:
yeah no problem

Reply:
Done!